src/app/static/src/app/store/root/getters.ts

```diff
@@ -1,12 +1,32 @@
 import {RootState} from "../../root";
 import {Getter, GetterTree} from "vuex";
+import {Warning} from "../../generated";
 
 interface RootGetters {
     isGuest: Getter<RootState, RootState>
+    warnings: Getter<RootState, RootState>
 }
 
 export const getters: RootGetters & GetterTree<RootState, RootState> = {
     isGuest: (state: RootState, getters: any) => {
         return state.currentUser == "guest";
+    },
+
+    warnings: (getters: any, rootGetters: any, rootState: RootState) => (stepName: string) => {
+
+        const warnings = {
+            modelOptions: rootState.modelOptions.warnings,
+            modelRun: rootState.modelRun.warnings,
+            modelCalibrate: rootState.modelCalibrate.warnings
+        }
+
+        const filterWarnings = (warnings: Warning[], stepName: string) =>
```

Comment:
This doesn't feel right to me - I think this function should accept a `stepName` (as implied by the argument's name!) rather than the name of a location. Then the `enum` could be private to this file/function.

Therefore, callers of this getter would specify the name of the step (e.g. `modelOptions`), the getter would look up the corresponding location (i.e. `model_options`), find all the warnings that specify that location, and return them.

Reply:
I struggle to understand your preference here, the function works perfectly well as intended. 

Caller of the getter can make a call such as `getters.warnings(Steps.modelOptions)`

Reply:
The enum lives in root state, this implementation is for root module. That's not external. It is also reused in tests.

Reply:
I did look at that, but thought (maybe incorrectly) they were translation keys and therefore not the most appropriate. And I assumed (again incorrectly?) that the warnings component would be embedded in the relevant steps rather than the stepper itself, and the step would know its name rather than `textKey`!

Reply:
That sounds good to me.